examples/plot_synthetic.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+"""
+============================================
+Synthetic dataset example with visualization
+============================================
+"""
+
+# %%
+from fairlearn.datasets._synthetic import (
+    SensitiveDatasetMaker,
+    SensitiveFeature,
+)
+from fairlearn.metrics import (
+    MetricFrame,
+    false_positive_rate,
+    true_positive_rate,
+    selection_rate,
+    count
+)
+import numpy as np
+from sklearn.metrics import accuracy_score, precision_score, recall_score
+from sklearn.model_selection import train_test_split
+from sklearn.tree import DecisionTreeClassifier
+
+rng = np.random.RandomState(42)
+
+gender_feature = SensitiveFeature('Gender', ['Man', 'Other', 'Unspecified', 'Woman'])
+age_feature = SensitiveFeature('Age', ['<21', '21-64', '>64'])
+
+dataset = SensitiveDatasetMaker(sensitive_features=[gender_feature, age_feature], random_state=rng)
+
+# need a convenience method here
+dataset.configured_groups[('Man', '<21')].classification_kwargs['n_samples'] = 100
```

Comment:
When reading this example I'm not entirely sure whether the other groups were forgotten or whether it will "just work" by having all the other groups with random sizes. Can you add a little bit of description (as @adrinjalali suggested in another comment)?

Reply:
After re-reading the code I realize now that all groups will have 300 rows except man<21 which will have 100. Originally, I had assumed that 300 is the total size and since man<21 gets 100 rows the remaining 200 are divided up into the other groups at random (or perhaps equally). This misunderstanding of mine shows that we'll need to be a bit clearer in explaining the functionality. I can see a few ways:
- rename the `n_samples` param because it's really `n_samples_per_group` OR change functionality (e.g., making that the total number of samples, but that's probably harder to achieve since the other way is already implemented)
- add a comment here and a nice docstring in the class explaining what's happening under the hood
- explain what happens to the groups that aren't manually configured
The bar plot below should show the group sizes, so that's great!

Reply:
I was also confused by this and I like @romanlutz's suggested solution!

Reply:
Changed to `n_samples_per_group`. 

> OR change functionality (e.g., making that the total number of samples, but that's probably harder to achieve since the other way is already implemented)

I did implement this in the previous iteration, but I've removed it here, just to reduce complexity. If you think it would be useful it wouldn't be hard to bring back in.

Reply:
I'm fine with `n_samples_per_group`. Would love to hear other @fairlearn/fairlearn-maintainers opinions, though!